torch/csrc/distributed/rpc/request_callback_impl.cpp

```diff
@@ -107,6 +112,7 @@ std::shared_ptr<FutureMessage> RequestCallbackImpl::processRpc(
       // scriptRemoteCall is only alive within this block, use reference to
       // avoid copy
       auto& stack = scriptRemoteCall.stackRef();
+      RRefContext::getInstance().waitForThreadLocalPendingUsers();
```

Comment:
Can we do this in an async way where we don't block the RPC thread? We can add callbacks on the futures in userTable_ to mark an 'uber' future as complete once done. Then, the 'uber' future can have a callback that actually does all the RPC execution itself. Once the RPC execution is done, we can mark the future that was returned to the RPC layer as complete so that it can respond back to the user.

Given we plan to merge RPC and JIT pools, we can't have blocking calls like these.

Reply:
Yes, this should work. let me try.